### PR TITLE
[Serve] Don't use Ray Jobs for local `serve run`

### LINKED ISSUE #32881: [serve] `serve run` command within job entrypoint fails if a `working_dir` is specified
Repro:
```
(anyscale) eoakes@Edwards-MacBook-Pro-2 tmp % ray --version
ray, version 2.3.0
(anyscale) eoakes@Edwards-MacBook-Pro-2 tmp % ray start --head
Usage stats collection is enabled. To disable this, add `--disable-usage-stats` to the command that starts the cluster, or run the following command: `ray disable-usage-stats` before starting the cluster. See https://docs.ray.io/en/master/cluster/usage-stats.html for more details.

Local node IP: 127.0.0.1

--------------------
Ray runtime started.
--------------------

Next steps
  To connect to this Ray runtime from another node, run
    ray start --address='127.0.0.1:6379'
  This Ray runtime only accepts connections from local host.
  To accept connections from remote hosts, specify a public ip when starting
  the head node: ray start --head --node-ip-address=<public-ip>.

  Alternatively, use the following Python code:
    import ray
    ray.init(address='auto')

  To see the status of the cluster, use
    ray status
  To monitor and debug Ray, view the dashboard at
    127.0.0.1:8265

  If connection fails, check your firewall settings and network configuration.

  To terminate the Ray runtime, run
    ray stop
(anyscale) eoakes@Edwards-MacBook-Pro-2 tmp % ray job submit --working-dir=https://github.com/anyscale/docs_examples/archive/refs/heads/main.zip -- serve run --non-blocking serve_hello:entrypoint
Job submission server address: http://127.0.0.1:8265

-------------------------------------------------------
Job 'raysubmit_CHgG77Qpg2fF7vum' submitted successfully
-------------------------------------------------------

Next steps
  Query the logs of the job:
    ray job logs raysubmit_CHgG77Qpg2fF7vum
  Query the status of the job:
    ray job status raysubmit_CHgG77Qpg2fF7vum
  Request the job to be stopped:
    ray job stop raysubmit_CHgG77Qpg2fF7vum

Tailing logs until the job exits (disable with --no-wait):
2023-02-27 17:56:15,704	INFO <string>:62 -- Deploying from import path: serve_hello:entrypoint.
Traceback (most recent call last):
  File "<string>", line 124, in <module>
  File "<string>", line 63, in main
  File "/Users/eoakes/miniforge3/envs/anyscale/lib/python3.8/site-packages/ray/_private/utils.py", line 1245, in import_attr
    module = importlib.import_module(module_name)
  File "/Users/eoakes/miniforge3/envs/anyscale/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 973, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'serve_hello'

------------------------------------------
Job 'raysubmit_CHgG77Qpg2fF7vum' succeeded
------------------------------------------
```

Using `python serve_hello.py` as the entrypoint instead works because this path goes over Ray client. This is likely caused by @zcin's recent change because we are no longer picking up the runtime_env of the job for the `serve run` command.

### PR DESCRIPTION
**Pull request by zcin**
## Why are these changes needed?

With the change from https://github.com/ray-project/ray/pull/30913, the local behavior of `serve run` changed. This PR reverts those changes. More details:

- Previously, if connecting to a local cluster, `serve run` will directly call `ray.init()` and `serve.run`
- With the above-mentioned change, even when connecting to a local cluster, `serve run` would submit a Ray Job to the local cluster and then call `ray.init()` and `serve.run()` inside that job. This caused the issue https://github.com/ray-project/ray/issues/32881, where if a user ran `serve run` inside a Job (or any other process with a runtime environment), they should expect to have access to the dependencies installed in that runtime environment. However since `serve run` goes through another layer of Job submission, they don't.

## Related issue number

Closes #32881

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by sihanwang41 — APPROVED** [review]

File: doc/source/serve/dev-workflow.md [hunk 1]
```diff
@@ -61,10 +61,6 @@ serve run local_dev:graph
 
 The `serve run` command blocks the terminal and can be canceled with Ctrl-C. Typically, `serve run` should not be run simultaneously from multiple terminals, unless each `serve run` is targeting a separate running Ray cluster.
 
-:::{note}
```
Comment (shrekris-anyscale) [inline doc/source/serve/dev-workflow.md:64]: Why are we removing this note? Do users no longer need to start Ray in the relevant directory?

File: python/ray/serve/run_script.py [hunk 2]
```diff
@@ -16,27 +16,22 @@
...
+    host: str,
+    port: int,
+    blocking: bool,
+    gradio: bool,
+):
     """
-    This is the Job that gets submitted to the Ray Cluster when `serve run` is executed.
-
     Loads the Serve app (either from a YAML config file or a direct import path), starts
     Serve and runs the app. By default, the code blocks until a SIGINT signal is
```
Comment (shrekris-anyscale) [inline python/ray/serve/run_script.py:29]:
```suggestion
    Serve, and runs the app. By default, the code blocks until a SIGINT signal is
```

**Review by shrekris-anyscale — COMMENTED** [review]
Nice work so far! I left some comments.

**Comment by edoakes** [comment]
@zcin do we still plan to merge this?

**Comment by zcin** [comment]
@edoakes Yes but it's pretty low priority right now

**Review by aslonnie — COMMENTED** [review]
is this still relevant? should we close this?

**Pull request closed without merging**